### PR TITLE
feat: hidden kanban groups

### PR DESCRIPTION
**Pull request by richardshiue**
### Feature Preview

---

#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database_view/board/application/board_bloc.dart [hunk 1]
```diff
@@ -410,7 +465,7 @@ class GroupItem extends AppFlowyGroupItem {
   GroupItem({
     required this.row,
     required this.fieldInfo,
-    bool draggable = true,
+    draggable = true,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/application/board_bloc.dart:468]: My bad
```suggestion
    bool draggable = true,
```

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart [hunk 2]
```diff
@@ -112,22 +113,31 @@ class _BoardColumnHeaderState extends State<BoardColumnHeader> {
...
+            height: 50,
+            child: Row(
+              children: [
+                _buildHeaderIcon(boardCustomData),
+                title,
+                const HSpace(6),
+                _groupOptionsButton(context),
+                const HSpace(4),
+                FlowyTooltip(
+                  message: "Add a new card at the top",
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart:128]: Localization

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart [hunk 3]
```diff
@@ -181,45 +190,96 @@ class _BoardColumnHeaderState extends State<BoardColumnHeader> {
...
+      case hide:
+        context
+            .read<BoardBloc>()
+            .add(BoardEvent.toggleGroupVisibility(group, false));
+        break;
+      // case GroupOptions.color:
+      //   break;
+      // case GroupOptions.delete:
+      //   // context.read<BoardBloc>().add(BoardEvent.);
+      //   break;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart:272]: Delete

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.fromLTRB(48, 16, 8, 8),
+                    child: Row(
+                      children: [
+                        Expanded(
+                          child: FlowyText.medium(
+                            'Hidden groups',
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart:60]: Localization

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+                  ),
+                ],
+              ),
+            ),
+    );
+  }
+
+  Widget _collapseExpandIcon() {
+    return FlowyTooltip(
+      message: isCollapsed ? "Expand group" : "Collpase group",
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart:83]: Localization

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+  final String viewId;
+  final FieldInfo primaryField;
+  final RowCache rowCache;
+  const HiddenGroupPopupItemList({
+    required this.bloc,
+    required this.viewId,
+    required this.primaryField,
+    required this.rowCache,
+    super.key,
+  });
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart:270]:
```suggestion
  const HiddenGroupPopupItemList({
    super.key,
    required this.bloc,
    required this.viewId,
    required this.primaryField,
    required this.rowCache,
  });

  final HiddenGroupButtonBloc bloc;
  final String viewId;
  final FieldInfo primaryField;
  final RowCache rowCache;
```

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+  final VoidCallback onPressed;
+  const HiddenGroupPopupItem({
+    super.key,
+    required this.cellContext,
+    required this.onPressed,
+    required this.cellBuilder,
+    required this.rowController,
+    required this.primaryField,
+    required this.renderHook,
+  });
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart:375]:
```suggestion
  const HiddenGroupPopupItem({
    super.key,
    required this.cellContext,
    required this.onPressed,
    required this.cellBuilder,
    required this.rowController,
    required this.primaryField,
    required this.renderHook,
  });

  final DatabaseCellContext cellContext;
  final FieldInfo primaryField;
  final RowController rowController;
  final CardCellBuilder cellBuilder;
  final RowCardRenderHook<String> renderHook;
  final VoidCallback onPressed;
```

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart [hunk 8]
```diff
@@ -181,45 +190,96 @@ class _BoardColumnHeaderState extends State<BoardColumnHeader> {
...
 
-  return widget;
+  FlowySvgData icon() => switch (this) {
+        rename => FlowySvgs.edit_s,
+        hide => FlowySvgs.hide_s,
+      };
+
+  String text() => switch (this) {
+        rename => "Rename",
+        hide => "Hide",
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart:283]: Localization

File: frontend/appflowy_flutter/lib/plugins/database_view/board/application/hidden_groups_bloc.dart [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+      onGroupChanged: GroupCallbacks(
+        onGroupByField: (List<GroupPB> groups) {
+          if (isClosed) return;
+
+          add(HiddenGroupsEvent.didReceiveHiddenGroups(groups: groups));
+        },
+        onUpdateGroup: (List<GroupPB> updatedGroups) {
+          if (isClosed) return;
+
+          final newGroups = List<GroupPB>.from(_groups);
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/application/hidden_groups_bloc.dart:57]:
```suggestion
          final newGroups = [..._groups];
```

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart [hunk 10]
```diff
@@ -181,45 +190,89 @@ class _BoardColumnHeaderState extends State<BoardColumnHeader> {
...
+      case hide:
+        context
+            .read<BoardBloc>()
+            .add(BoardEvent.toggleGroupVisibility(group, false));
+        break;
+    }
   }
 
-  return widget;
+  FlowySvgData icon() => switch (this) {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart:269]: get icon =>

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart [hunk 11]
```diff
@@ -181,45 +190,89 @@ class _BoardColumnHeaderState extends State<BoardColumnHeader> {
...
+        rename => FlowySvgs.edit_s,
+        hide => FlowySvgs.hide_s,
+      };
+
+  String text() => switch (this) {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_column_header.dart:274]: get text =>

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart [hunk 12]
```diff
@@ -0,0 +1,398 @@
...
+                child: Padding(
+                  padding:
+                      const EdgeInsets.symmetric(horizontal: 6, vertical: 4),
+                  child: Row(
+                    children: [
+                      // Opacity(
+                      //   opacity: isHovering ? 1 : 0,
+                      //   child: const HiddenGroupCardActions(),
+                      // ),
+                      // const HSpace(4),
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/widgets/board_hidden_groups.dart:211]: remove it.

**Pull request merged**